scripts/un/energy/un_energy_codes.py

```diff
@@ -0,0 +1,1431 @@
+        code = code[1:]
+    elif add_pv_from_map(code[:1], UN_ENERGY_PLANT_TYPE, pv, counters):
+        code = code[1:]
+
+    if len(code) > 0:
+        # Suffix of the producer code not mapperd to any property.
+        _add_error_counter(
+            f'warning_ignored_producer_code_{code}',
+            f'Ignored producer code sufffix: {code} in {orig_code}', counters)
+    return True
+
+
+def add_pv_for_consumption_code(code: str, pv: dict, counters=None) -> bool:
+    """Adds PVs for energy consumption code into the pv dict.
+
+    The consumption code is prefixed as follows:
+      '1': Manufacturing
+      '2': Transport
+      '3': others
+
+    Args:
+      code: string for the UN Energy code for energy consumption
+      pv: dictionary of PVs to be updated
+      counters: [optional] error counters to be updated
+
+    Returns:
+      True if any properties were added into the pv.
+    """
+    orig_code = code
+    if not code.startswith('12'):
+        return False
+    code = remove_prefix(code, '12')
+    pv['measuredProperty'] = 'dcs:consumption'
+    if len(code) > 0:
+        prefix = add_pv_from_map_for_prefix(code, UN_ENERGY_CONSUMING_INDUSTRY,
+                                            pv, counters)
+        if prefix is not None:
+            code = remove_prefix(code, prefix)
+    if len(code) > 0:
+        _add_error_counter(
+            f'error_ignored_consumption_code_{code}',
+            f'Ignored consumption code: {orig_code}, suffix: {code}', counters)
+    return True
+
+
+def add_pv_for_capacity_code(code: str, pv: dict, counters=None) -> bool:
+    """Adds PVs for energy capacity code into the pv dict.
+
+    Args:
+      code: UN energy capacity code string
+      pv: dictionary of PVs to be updated
+      counters: [optional] error counters to be updated
+    
+    Returns:
+      True if any properties were added into the pv.
+    """
+    orig_code = code
+    if not code.startswith('13'):
+        return False
+    code = remove_prefix(code, '13')
+    pv['measuredProperty'] = 'dcs:capacity'
+    prefix = add_pv_from_map_for_prefix(code, UN_ENERGY_CAPACITY_CODES, pv,
+                                        counters)
+    if prefix is None:
+        _add_error_counter('error_ignored_capacity_code',
+                           f'Ignored energy capacity code {code}', counters)
+        return False
+    code = remove_prefix(code, prefix)
+    if len(code) > 0:
+        # Add plant type
+        prefix = add_pv_from_map_for_prefix(code, UN_ENERGY_CAPACITY_PLANT_CODE,
+                                            pv, counters)
+        if prefix is not None:
+            code = remove_prefix(code, code)
+
+    if len(code) > 0:
+        _add_error_counter(
+            f'error_ignored_capacity_code_{code}',
+            f'Ignored capacity code: {orig_code}, suffix: {code}', counters)
+    return True
+
+
+def add_pv_for_property(code: str,
```

Comment:
this seems complicated :)

Reply:
Easier than defining a map with 200+ codes :)